I2C cleanup to remove usage of CONFIG_I2C_x_DEFAULT_CFG

galak opened the pull request:
This reworks things based on comments from PR #722.  We now get the initial bitrate from device tree.  We also for now always default to MASTER mode since that is all we really support (and I didn't want to introduce a solution to master/slave in dt right now).

galak committed:
dts: nxp: Fix formatting of fixup files

Cleanup whitespace so things align and are easier to read.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
i2c: Cleanup dts clock-frequency property usage

Clarify that the clock-frequency is the bitrate at boot and introduce
defines that .dts files can use to set the clock-frequency.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
i2c: mcux: Remove usage of CONFIG_I2C_x_DEFAULT_CFG

Stop using CONFIG_I2C_x_DEFAULT_CFG to get the initial value.  Since we
only support master mode we always default to it for initial config and
we get the bitrate from the device tree.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
i2c: stm32: Remove usage of CONFIG_I2C_x_DEFAULT_CFG

Stop using CONFIG_I2C_x_DEFAULT_CFG to get the initial value.  Since we
only support master mode we always default to it for initial config and
we get the bitrate from the device tree.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
i2c: Add helper function to map from bitrate values to zephyr cfg

Add _i2c_map_dt_bitrate() that maps from a bitrate to the I2C zephyr
cfg bit fields used in i2c_configure().

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

ydamigos commented on dts/arm/disco_l475_iot1.fixup:
```diff
@@ -42,10 +42,12 @@
 #define CONFIG_I2C_1_NAME			ST_STM32_I2C_V2_40005400_LABEL
 #define CONFIG_I2C_1_EVENT_IRQ			ST_STM32_I2C_V2_40005400_IRQ_EVENT
 #define CONFIG_I2C_1_ERROR_IRQ			ST_STM32_I2C_V2_40005400_IRQ_ERROR
+#define CONFIG_I2C_1_BITRATE			ST_STM32_I2C_V1_40005400_CLOCK_FREQUENCY
```
It should be ST_STM32_I2C_V2_40005400_CLOCK_FREQUENCY

galak replied:
Good catch, fixed now.

ydamigos reviewed (commented):
I tested it on 96b_carbon and works fine. Just a typo to be fixed.

nashif commented on include/i2c.h:
```diff
@@ -55,6 +56,24 @@ extern "C" {
 /** Controller to act as Master. */
 #define I2C_MODE_MASTER			(1 << 4)
 
+static inline int __i2c_map_dt_bitrate(u32_t bitrate)
```
does this belong in the i2c API header? why not have this in dt-bindings where other defines used by the function also exist?

galak replied:
The dt-bindings headers should try and be standalone (so avoid Zephyr specific defines or code there).  I can move this into a private drivers/i2c/i2c-priv.h header

mnkp replied:
Kernel is using single underscore to indicate private function, here we have two.

galak replied:
fixed

mnkp commented on drivers/i2c/i2c_ll_stm32.c:
```diff
@@ -90,13 +90,15 @@ static int i2c_stm32_init(struct device *dev)
 	clock_control_on(clock, (clock_control_subsys_t *) &cfg->pclken);
 
-	ret = i2c_stm32_runtime_configure(dev, data->dev_config.raw);
+	bitrate_cfg = __i2c_map_dt_bitrate(cfg->bitrate) << 1;
```
We should probably add 
```
#define I2C_SPEED_SHIFT 1
```
to include/i2c.h file and have here
```
bitrate_cfg = __i2c_map_dt_bitrate(cfg->bitrate) << I2C_SPEED_SHIFT;
```
or even better __i2c_map_dt_bitrate() could do the shift.

galak replied:
Yeah, I wasn't sure what made sense, but probably doing the shift in the function makes sense.

galak commented on include/dt-bindings/i2c/i2c.h:
```diff
@@ -0,0 +1,15 @@
+#ifndef __DT_BINDINGS_I2C_I2C_H
+#define __DT_BINDINGS_I2C_I2C_H
+
+#define I2C_BITRATE_STANDARD	100000	/* 100 K */
```
fixed

galak commented on dts/arm/nxp/nxp_k6x.dtsi:
```diff
@@ -76,6 +77,7 @@
 
 		i2c0: i2c@40066000 {
 			compatible = "nxp,kinetis-i2c";
+			clock-frequency = <I2C_BITRATE_STANDARD>;
```
I think its better to have a valid setting

galak commented on dts/arm/frdm_kl25z.fixup:
```diff
@@ -1,12 +1,13 @@
+#define CONFIG_ADC_0_BASE_ADDRESS		NXP_KINETIS_ADC16_4003B000_BASE_ADDRESS_0
+#define CONFIG_ADC_0_IRQ			NXP_KINETIS_ADC16_4003B000_IRQ_0
+#define CONFIG_ADC_0_IRQ_PRI			NXP_KINETIS_ADC16_4003B000_IRQ_0_PRIORITY
+#define CONFIG_ADC_0_NAME			NXP_KINETIS_ADC16_4003B000_LABEL
```
it was intended, there's a patch to just adjusts whitespace.

mnkp commented on drivers/i2c/i2c-priv.h:
```diff
@@ -0,0 +1,39 @@
+{
+	switch (bitrate) {
+	case I2C_BITRATE_STANDARD:
+		return I2C_SPEED_STANDARD << 1;
```
I think we should still add
```
#define I2C_SPEED_SHIFT 1
```
to include/i2c.h file and use it here rather than hard coding the value.

galak replied:
done

mnkp commented on drivers/i2c/i2c-priv.h:
```diff
@@ -0,0 +1,39 @@
+	case I2C_BITRATE_ULTRA:
+		return I2C_SPEED_ULTRA << 1;
+	}
+
```
Maybe it's worth to add here:
```
SYS_LOG_ERR("Invalid I2C bit rate value");
```

galak replied:
done

mnkp reviewed (commented):
This PR looks all good to me. I have only a minor remark.

MaureenHelm commented on include/dt-bindings/i2c/i2c.h:
```diff
@@ -0,0 +1,15 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef __DT_BINDINGS_I2C_I2C_H
+#define __DT_BINDINGS_I2C_I2C_H
```
Is the double I2C on purpose?

galak replied:
yes, meant to map to the dir path + file name.

MaureenHelm reviewed (commented):
There's a typo in the last two commit messages:
"we always default to for initial config"

Otherwise LGTM

galak committed:
i2c: Add define for bit position/shift of I2C SPEED settings in config

Added a define to use in code that provides the amount we need to shift
the speed settings in the i2c config params.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>
